cmd/govim/gopls.go

```diff
@@ -20,16 +20,20 @@ import (
 )
 
 func (g *govimplugin) startGopls() error {
-	logfile, err := g.createLogFile("gopls")
-	if err != nil {
-		return err
-	}
-	logfile.Close()
-	g.Logf("gopls log file: %v", logfile.Name())
+	goplsArgs := []string{"-rpc.trace"}
+	if getEnvVal(g.goplsEnv, config.EnvLog, "on") == "on" {
+
+		logfile, err := g.createLogFile("gopls")
+		if err != nil {
+			return err
+		}
+		logfile.Close()
```

Comment:
I would've expected a `defer` here, but I see there was no `defer` in the old code, so perhaps it's intentionally absent. If that's the case, please pardon my ignorance.

Reply:
It's a good question. All we need to do here is create the log file; we, `govim`, never write to this log file. That's `gopls`' job. So we just close our file handle (to tidy up) and then forget about the file. The reason we, `govim`, create the logfile is so that we can report to the user where `gopls` is logging, rather than have `gopls` create a random temporary file of its own (there is no protocol for it telling us where the logfile is, hence we "inject" the location).

Reply:
Makes perfect sense then. Thanks for the explanation.